bot_handler.py

```diff
@@ -74,42 +82,93 @@ def on_bot_update(self, update):
-            self.send_message(f'Ви додали до замовлення:\n{update.message.text}', COMPLETE_ORDER_BUTTON)
-            
+        if update.message.text != COMPLETE_ORDER_BUTTON:                             
+            current_item = str(update.message.text)
+            print(f'curr item: {current_item}')    
+
+            user_data = self.read_user_data()
+            try:      
+                user_data['order']['item'].append(current_item)
+            except:
+                user_data['order']['item'] = []
+                user_data['order']['item'].append(current_item)
+
+            self.write_user_data(user_data['contact'], user_data['order'])  
+
+            #self.get_user_data()['order']['item'].append(current_item)
+            self._next_state = RequestState2.name
+         
         elif update.message.text == COMPLETE_ORDER_BUTTON:
             self._next_state = CompleteOrderState.name
 
 
+class RequestState2(HandlerState):
+    name = 'RequestState2'
+
+    def enter(self):
+        super().enter()
+
+        self.send_message('Введіть кількість')
+
+    def on_bot_update(self, update):
+        global current_quantity
+
+        current_quantity = str(update.message.text)
+        print(f'curr qunat: {current_quantity}')
+
+        user_data = self.read_user_data()
+
+        try:      
+            user_data['order']['quantity'].append(current_quantity)
+        except:
+            user_data['order']['quantity'] = []
+            user_data['order']['quantity'].append(current_quantity)
+        self.write_user_data(user_data['contact'], user_data['order']) 
+
+        self.send_message(f'Ви додали до замовлення:\n{current_item} у кількості {current_quantity}', COMPLETE_ORDER_BUTTON)
+        self._next_state = RequestState.name
+
 class CompleteOrderState(HandlerState):
     name = 'CompleteOrderState'
+    user_data = {}
+    order = None
 
     def enter(self):
         super().enter()
+        self.user_data = self.read_user_data()
+        order_dict = self.user_data['order']
+        print(order_dict)
 
-        order = self.get_user_data()['order']
-
-        self.send_message(f'Ваше замовлення:{NEW_LINE}{NEW_LINE.join(str(item) for item in order)}', CONFIRM_ORDER_BUTTON, CANCEL_ORDER_BUTTON)
+        self.order = list(zip(*order_dict.values()))
+        print(self.order)
+        
+        self.send_message(f'Ваше замовлення:{NEW_LINE}{NEW_LINE.join(str(item[0] + PH + item[1]) for item in self.order)}', 
+                            CONFIRM_ORDER_BUTTON, CANCEL_ORDER_BUTTON)
 
     def on_bot_update(self, update):  
         if update.message.text == CONFIRM_ORDER_BUTTON:
             self.send_message('Замовлення підтверджено')
-            #further order processing
-            #...
-            #...
-            #...
+            for item, amount in self.order:
+                send_to_notion('db_id_placeholder', item, amount, self.user_data['contact'])
```

Comment:
As you can see in `notion_requests` module, method `create_page` awaits for `POST` request to be done. This means that `send_to_notion` calls will also await for the request to be done. I suggest that we send current user somewhat of a success message after all `send_to_notion` calls. Please see suggestion below.

Reply:
But when it comes to improvement I would say that we'll need to change `notion_requests` module behaviour so we don't have to wait for request to be completed (we should use async behaviour instead). When we do so, we put all our users in unseen queue, where we can't serve user2 while we are making requests for user1.

It is just thoughts, nothing needs to be done on this cause for now.